File: app/src/main/java/com/lucas/omnia/adapters/SearchLawAdapter.java
package com.lucas.omnia.adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.lucas.omnia.R;
import com.lucas.omnia.activities.LawPageActivity;
import com.lucas.omnia.models.Law;

import java.util.List;

public class SearchLawAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private final List<Law> lawList;
    private final Context context;

    public SearchLawAdapter(Context context, List<Law> lawList) {
        this.context = context;
        this.lawList = lawList;
    }

    public class LawViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        TextView localeTextView;
        TextView authorityTextView;
        TextView titleTextView;
        TextView descriptionTextView;

        LawViewHolder(View itemView) {
            super(itemView);

            localeTextView = itemView.findViewById(R.id.law_tv_locale_body);
            authorityTextView = itemView.findViewById(R.id.law_tv_authority_body);
            titleTextView = itemView.findViewById(R.id.law_tv_title_body);
            descriptionTextView = itemView.findViewById(R.id.law_tv_description_body);

            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            int position = getAdapterPosition();
            if (position != RecyclerView.NO_POSITION) {
                Law law = lawList.get(position);

                Intent intent = new Intent(context, LawPageActivity.class);
                intent.putExtra(LawPageActivity.EXTRA_LAW_URN, law.getUrn());
                context.startActivity(intent);
            }
        }
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        return new LawViewHolder(inflater.inflate(R.layout.item_law, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int position) {
        Law law = lawList.get(position);
        TextView localeTextView = ((LawViewHolder) viewHolder).localeTextView;
        TextView authorityTextView = ((LawViewHolder) viewHolder).authorityTextView;
        TextView titleTextView = ((LawViewHolder) viewHolder).titleTextView;
        TextView descriptionTextView = ((LawViewHolder) viewHolder).descriptionTextView;

        localeTextView.setText(law.getLocale());
        authorityTextView.setText(law.getAuthority());
        titleTextView.setText(law.getTitle());
        descriptionTextView.setText(law.getDescription());
    }

    @Override
    public int getItemCount() {
        return lawList.size();
    }
}

File: app/src/main/java/com/lucas/omnia/adapters/SearchLawAdapter.kt
package com.lucas.omnia.adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.lucas.omnia.R
import com.lucas.omnia.activities.LawPageActivity
import com.lucas.omnia.models.Law

class SearchLawAdapter(private val context: Context, private val lawList: List<Law>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    inner class LawViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        var localeTextView: TextView = itemView.findViewById(R.id.law_tv_locale_body)
        var authorityTextView: TextView = itemView.findViewById(R.id.law_tv_authority_body)
        var titleTextView: TextView = itemView.findViewById(R.id.law_tv_title_body)
        var descriptionTextView: TextView = itemView.findViewById(R.id.law_tv_description_body)
        override fun onClick(view: View) {
            val position = adapterPosition
            if (position != RecyclerView.NO_POSITION) {
                val (urn) = lawList[position]
                val intent = Intent(context, LawPageActivity::class.java)
                intent.putExtra(LawPageActivity.EXTRA_LAW_URN, urn)
                context.startActivity(intent)
            }
        }

        init {
            itemView.setOnClickListener(this)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return LawViewHolder(inflater.inflate(R.layout.item_law, parent, false))
    }

    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, position: Int) {
        val (_, locale, authority, title, description) = lawList[position]
        val localeTextView = (viewHolder as LawViewHolder).localeTextView
        val authorityTextView = viewHolder.authorityTextView
        val titleTextView = viewHolder.titleTextView
        val descriptionTextView = viewHolder.descriptionTextView
        localeTextView.text = locale
        authorityTextView.text = authority
        titleTextView.text = title
        descriptionTextView.text = description
    }

    override fun getItemCount(): Int {
        return lawList.size
    }
}